### DISCUSSION TITLE
v1.13.9a

### DISCUSSION THREAD

**Discussion by deividAlfa** [deividAlfa/stm32_soldering_iron_controller#119] [Announcements]
- Perform manual flash sector size detection instead of relying on the reported size by the Flash register.
  This should fix the fake/clones that were working with previous versions.

- Fix bug in calibration settings screen where the PID would react when switching between Zero set/Sampling/Captured.
  This would cause a small power pulse into the tip, warming it up.
  
- If the oled poweroff option is enabled, keep it on until the tip is colder than 60ºC.

<hr /><em>This discussion was created from the release <a href='https://github.com/deividAlfa/stm32_soldering_iron_controller/releases/tag/v1.13.9a'>v1.13.9a</a>.</em>

**Comment by jopl** [comment]
Dear Dave,
thank you againn so much for your work!
Just a subject to improvement ... I am thinking about incresing number of heating elements (for soldering tweezers, which have in my case 2 of them, working independently).
I drawn the schematic as well as PCB (the same dimensions, different connector /aviatic G16 - 7 pins/) to be able to test it (attached). I am trying to be as compatible as possible.
My intention is to have new profile for tweezers, selectable in the menu, which will be able to control 2 PWMs and read 2 thermocouples. All settings will be the same for both heating elements.
Do you think it is easily possible to integrate into the current firmware, please?
Untill now, I tested both PWMs (CH1 and CH3) are able to work together as well as 2 ADC INs.
Waiting for the PCB production at the moment.
Thx for reply in advance.
Regards, Josef.

[Soldering_Tweezers_10_Schematics.pdf](https://github.com/deividAlfa/stm32_soldering_iron_controller/files/14551175/Soldering_Tweezers_10_Schematics.pdf)
![Soldering_Tweezers_Driver_10](https://github.com/deividAlfa/stm32_soldering_iron_controller/assets/401876/b9669dc7-5357-418f-95aa-77c3177f83bc)

**Comment by deividAlfa** [comment]
Of course it can be done, but my project only targets "official" controllers, not everyone's mods.
Looks nice, but please understand I can't make custom firmware for everyone, this should be your own work and your own fork, sorry!
